include/pipettes/core/tasks/motion_controller_task.hpp

```diff
@@ -117,9 +154,52 @@ class MotionControllerMessageHandler {
         controller.send_usage_data(m.message_index, usage_client);
     }
 
+    void handle(const pipettes::task_messages::motor_control_task_messages::
+                    RouteMotorDriverInterrupt& m) {
+        if (m.debounce_count > 9) {
+            if (controller.read_tmc_diag0()) {
+                controller.stop(
+                    can::ids::ErrorSeverity::unrecoverable,
+                    can::ids::ErrorCode::motor_driver_error_detected);
+                if (!controller.is_timer_interrupt_running()) {
+                    can_client.send_can_message(
+                        can::ids::NodeId::host,
+                        can::messages::ErrorMessage{
+                            .message_index = m.message_index,
+                            .severity = can::ids::ErrorSeverity::unrecoverable,
+                            .error_code = can::ids::ErrorCode::
+                                motor_driver_error_detected});
+                    driver_client.send_motor_driver_queue(
+                        can::messages::ReadMotorDriverErrorStatusRequest{
+                            .message_index = m.message_index});
+                }
+            } else {
+                controller.clear_cancel_request();
```

Comment:
Do we need this? If we're here, then this is the first time that we've interacted with this message, and therefore we haven't set a cancel request yet. It kind of seems like this would either (a) do nothing because there's no cancel request or (b) clear a cancel request from somewhere else which we definitely don't want.

Reply:
Good call!